corehq/apps/locations/views.py

```diff
@@ -373,8 +377,30 @@ def _attach_sync_boundaries_to_location_type(loc_type_data, loc_type_db):
                                                   if include_without_expanding_id else None)
         except KeyError:        # include_without_expanding location type was deleted
             loc_type.include_without_expanding = None
+        include_only = LocationTypesView._get_include_only(include_only_ids, loc_type_db)
+        loc_type.include_only.set(include_only)
         loc_type.save()
 
+    @staticmethod
+    def _get_include_only(include_only_ids, loc_type_db):
+        """The user specified that we include loc types `include_only_ids`, but
+        we need to insert any parent location types"""
```

Comment:
It would arguably be a nicer UX to do this in javascript, but this feels safer, and this is an advanced-user-only workflow